src/IB/IBHierarchyIntegrator.cpp

```diff
@@ -415,6 +415,16 @@ IBHierarchyIntegrator::initializePatchHierarchy(Pointer<PatchHierarchy<NDIM> > h
     // Finish Lagrangian data movement.
     d_ib_method_ops->endDataRedistribution(hierarchy, gridding_alg);
 
+    // Initialize composite hierarchy data that was not initailized by
+    // gridding algorithm's call to initializeLevelData(), which may depend on
+    // Lagrangian data.
+    const bool from_restart = RestartManager::getManager()->isFromRestart();
+    const bool initial_time = MathUtilities<double>::equalEps(d_integrator_time, d_start_time);
+    if (!from_restart)
```

Comment:
Only parent integrator should make this call. 

Something like,
if (d_parent_integrator == NULL && !from_restart)

Does this make sense?

Reply:
Although this looks like an extra check. In practice initializePatchHierarchy() and regridHierarchy() is called only by the top-most integrator. So, it looks fine. 

@boyceg What do you say?